test/other/embind_jsgen_method_pointer.cpp

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <fstream>
+#include <emscripten.h>
+#include <emscripten/bind.h>
+
+using namespace emscripten;
+
+class Foo {
+public:
+  void go(const std::string& filename) {
+    std::ifstream ifs(filename, std::ifstream::binary);
+    ifs.close();
+  }
+};
+
+int main() {
+  printf("done\n");
+}
+
+EMSCRIPTEN_BINDINGS(xxx) {
+    class_<Foo>("Foo")
+      .function("go", &Foo::go);
```

Comment:
Is this different to how method are normally install?  i.e. is taking a pointer to a method not already done in lots of other tests?

Reply:
It's the normal way. The important part is `std::ifstream` which seems to trigger a bunch of filesystem stuff being included and causes the issue with the allocations not being in the same order anymore.

Reply:
Oh, in that case I wonder if we can be more explicitly and precise here.  Can we create a static ctor here that calls malloc?   And I guess you should rename it back something that mentions static ctors... sorry for the churn